src/MainWindow.vala

```diff
@@ -998,18 +998,18 @@ namespace Terminal {
         }
 
         private void on_get_text (Gtk.Clipboard board, string? intext) {
-            /* if unsafe paste alert is enabled, show dialog */
-            if (Application.settings.get_boolean ("unsafe-paste-alert") && !unsafe_ignored ) {
+            if (intext == null) {
+                return;
+            }
 
-                if (intext == null) {
-                    return;
-                }
-                if (!intext.validate ()) {
-                    warning ("Dropping invalid UTF-8 paste");
-                    return;
-                }
-                var text = intext.strip ();
+            if (!intext.validate ()) {
+                debug ("Dropping invalid UTF-8 paste");
+                return;
+            }
+
+            var text = intext.strip ();
```

Comment:
```suggestion
            unowned string text = intext._strip ();
```
That way no copy of the input string (which can potentially be huge :man_shrugging:)